controllers/volumereplicationgroup_controller.go

```diff
@@ -256,15 +265,15 @@ func filterPVC(mgr manager.Manager, pvc *corev1.PersistentVolumeClaim, log logr.
 	//   to which the the pvc belongs to.
 	// - whether the labels on pvc match the label selectors from
 	//    VolumeReplicationGroup CR.
-	err := mgr.GetClient().List(context.TODO(), &vrgs, listOptions...)
```

Comment:
Why are we shifting to the reader here?

A cached VRG if it refers to the PVC the event would be processed and vrg reconciled. If a cached VRG does not refer to the PVC, then a future update to the cache will reconcile the vrg.

IOW, a VRG need not be read from the API server in these cases (to reduce request load to the same), what am I missing here?

Reply:
Moving the PVC mapping function out of `SetupWithManager` resulted in the `Manager` no longer being in context.  There was a conflict with the `client` name:
https://github.com/RamenDR/ramen/blob/b2076d473da4a3bfcd3bc6cab12cbf8139468213/controllers/volumereplicationgroup_controller.go#L258
and only a `Reader` was needed.  It is still using the _cached_ reader, getting it from:
https://github.com/RamenDR/ramen/blob/b2076d473da4a3bfcd3bc6cab12cbf8139468213/main.go#L137
and not 
https://github.com/RamenDR/ramen/blob/b2076d473da4a3bfcd3bc6cab12cbf8139468213/main.go#L138